orchestrator/websocket/managers/broadcast_websocket_manager.py

```diff
@@ -57,29 +58,54 @@ async def disconnect(
 
     async def disconnect_all(self) -> None:
         for websocket in self.connected:
-            await self.disconnect(websocket)
+            await self.disconnect(websocket, code=status.WS_1001_GOING_AWAY, reason="Shutting down")
 
     async def receiver(self, websocket: WebSocket, channel: str) -> None:
+        """Read messages from websocket client."""
         async for message in websocket.iter_text():
             logger.debug("Received websocket message", client=websocket.client, channel=channel, message=repr(message))
             if message == "__ping__":
                 await websocket.send_text("__pong__")
 
     async def sender(self, websocket: WebSocket, channel: str) -> None:
-        async with self.sub_broadcast.subscribe(channel=channel) as subscriber:
-            async for event in subscriber:
-                await websocket.send_text(event.message)
+        """Read messages from redis channel and send to websocket client."""
+        log = logger.bind(client=websocket.client, channel=channel)
 
-                json = json_loads(event.message)
+        def parse_message(raw_message: Any) -> str | None:
+            match raw_message:
+                case {"type": "message", "data": bytes() as data}:
+                    return data.decode()
+                case None:
+                    return None
+                case _:
+                    log.info("Drop unrecognized message", raw=raw_message)
+                    return None
+
+        async with self.broadcast.subscriber(channel) as subscriber:
+            log.debug("Websocket client subscribed to channel")
+            while True:
+                raw = await subscriber.get_message(timeout=1)
+                if (message := parse_message(raw)) is None:
+                    continue
+
+                log.debug("Send websocket message", message=message)
+                await websocket.send_text(message)
+
+                # Not sure if/where this is used.. will keep it for now.
+                json = json_loads(message)
```

Comment:
This can be removed. we used it at first to close the connection after the subscription detail page was complete.

Reply:
Thanks, I removed it and it seems to make no difference. The `websocket.disconnect()` being done in `disconnect_all()` is probably sufficient.